example-scala/src/main/scala/com/softwaremill/kmq/example/EmbeddedStream.scala

```diff
@@ -0,0 +1,114 @@
+package com.softwaremill.kmq.example
+
+import com.softwaremill.kmq.redelivery.streams.RedeliveryTracker
+import com.softwaremill.kmq.{KafkaClients, KmqClient, KmqConfig}
+import com.typesafe.scalalogging.StrictLogging
+import net.manub.embeddedkafka.{EmbeddedKafka, EmbeddedKafkaConfig}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.{ByteBufferDeserializer, ByteBufferSerializer}
+
+import java.nio.ByteBuffer
+import java.time.Duration
+import java.util.Random
+import java.util.concurrent.{ConcurrentHashMap, Executors}
+import scala.jdk.CollectionConverters.IterableHasAsScala
+
+object EmbeddedStream extends StrictLogging {
+  private val TOTAL_MSGS = 4
+  private val FAIL_RATIO = 0.5
+  private val PARTITIONS = 1
+
+  private implicit val kafkaConfig: EmbeddedKafkaConfig = EmbeddedKafkaConfig.defaultConfig
+  private implicit val clients: KafkaClients = new KafkaClients("localhost:" + kafkaConfig.kafkaPort)
+  private implicit val kmqConfig: KmqConfig = new KmqConfig("queue", "markers", "kmq_client", "kmq_redelivery",
+    Duration.ofSeconds(3).toMillis, 1000)
+
+  private val random: Random = new Random(0)
+  private val processedMessages = new ConcurrentHashMap[Integer, Integer]
+
+  final def main(args: Array[String]): Unit = {
+    EmbeddedKafka.start()
+    // The offsets topic has the same # of partitions as the queue topic.
+    EmbeddedKafka.createCustomTopic(kmqConfig.getMarkerTopic, partitions = PARTITIONS)
+    EmbeddedKafka.createCustomTopic(kmqConfig.getMsgTopic, partitions = PARTITIONS)
+    logger.info("Kafka started")
+
+    val redelivery = RedeliveryTracker.start()
+    sleep(1000) // Wait for the stream to warm up TODO: analyze
```

Comment:
did something stop working without this sleep?

Reply:
Without this sleep the stream does not process following messages. As if the stream didn't see messages before the start.

Reply:
maybe it's because of the consumer settings?